### PR TITLE
feat(scheduling): add e2e skeleton for volume level soft anti affinity

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#5358](https://github.com/longhorn/longhorn/issues/5358)

### CODE REVIEW

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]
If the manual cases are already covered in the automation test, no need to add the manual cases.

File: manager/integration/tests/test_scheduling.py [hunk 1]
```diff
@@ -1497,3 +1497,63 @@ def test_replica_schedule_to_disk_with_most_usable_storage(client, volume_name,
...
+
+    Then
+    - Verify volume will be in degraded state.
+    - Verify volume reports condition `scheduled == false`
+    - Verify only two of three replicas of volume are healthy.
+    - Verify the remaining replica doesn't have `replica.HostID`,
+      meaning it's unscheduled
+    - Check volume `data`
+    """
+    pass
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply manager/integration/tests/test_scheduling.py:1559]: Hi @khushboo-rancher 
We already have global soft_anti_affinity setting test
https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_scheduling.py#L95

- `def test_soft_anti_affinity_scheduling(client, volume_name)`: this is for **enable** global soft_anti_affinity setting
- `def test_hard_anti_affinity_scheduling(client, volume_name)`: this is for **disable** global soft_anti_affinity setting

In this PR, I added volume level test
- `def test_soft_anti_affinity_scheduling_volume_enable()`: **disable** global soft_anti_affinity setting, but **enable** volume soft_anti_affinity setting
- `def test_soft_anti_affinity_scheduling_volume_disable()`: **enable** global soft_anti_affinity setting, but **disable** volume soft_anti_affinity setting

thanks!

File: docs/content/manual/functional-test-cases/scheduling.md [hunk 2]
```diff
@@ -9,6 +9,15 @@ Manual Test
...
+
+
+| **Test name** | **Prerequisite** | **Expectation** |
+| --- | --- | --- |
+| EKS across zone scheduling - disable volume replicaZoneSoftAntiAffinity | **Prerequisite**:<br><br>*   EKS Cluster with 3 nodes across two AWS zones (zone#1, zone#2)<br><br>* Enable global replica zone anti-affinity <br><br>1.  Create a volume with 2 replicas, `replicaZoneSoftAntiAffinity=disabled` and attach it to a node.<br>2.  Delete a replica scheduled to each zone, repeat it few times<br>3.  Scale volume replicas to 3<br>4.  edit Volume spec to `replicaZoneSoftAntiAffinity=enabled` | *   Volume replicas should be scheduled one per AWS zone<br>*   Deleting a replica in a zone should trigger a replica rebuild<br>*   New rebuilding replica should be scheduled to the same zone as the deleted replica<br>*   Scaling volume replicas to 3, the new replica will not be scheduled to any node<br>*  The new replica will be scheduled after the setting is enabled |
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply docs/content/manual/functional-test-cases/scheduling.md]: Fixed, thanks!

**Review by innobead — CHANGES_REQUESTED** [review]
`test-the-trim-related-option-update-for-old-volumes.md` file name is not right.

File: docs/content/manual/release-specific/v1.5.0/test-the-trim-related-option-update-for-old-volumes.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+**Given** EKS Cluster with 3 nodes across 2 AWS zones (zone#1, zone#2)
+
+*And* Deploy Longhorn v1.5.0
+
+*And* Disable global replica zone anti-affinity
+
+*And* Create a volume with 2 replicas, `replicaZoneSoftAntiAffinity=enabled` and attach it to a node.
+
+*And* Upgrade Longhorn to the latest.
```
Comment (innobead) [inline docs/content/manual/release-specific/v1.5.0/test-the-trim-related-option-update-for-old-volumes.md:18]: What does this mean?

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one question.

File: docs/content/manual/release-specific/v1.5.0/test-the-trim-related-option-update-for-old-volumes.md (hunk 3 above)
Comment (ChanYiLin) [reply docs/content/manual/release-specific/v1.5.0/test-the-trim-related-option-update-for-old-volumes.md]: Fixed, thanks!

**Comment by ChanYiLin** [comment]
> `test-the-trim-related-option-update-for-old-volumes.md` file name is not right.

fixed, thanks

**Review by innobead — APPROVED** [review]

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**